src/App/DocumentObject.h

```diff
@@ -102,17 +104,13 @@ class AppExport DocumentObject: public App::TransactionalObject
 
     PropertyString Label;
     PropertyString Label2;
-    PropertyExpressionEngine ExpressionEngine;
+    PropertyExpressionEngine* ExpressionEngine;
 
     /// Allow control visibility status in App name space
     PropertyBool Visibility;
 
-    /// signal before changing a property of this object
-    boost::signals2::signal<void (const App::DocumentObject&, const App::Property&)> signalBeforeChange;
-    /// signal on changed  property of this object
-    boost::signals2::signal<void (const App::DocumentObject&, const App::Property&)> signalChanged;
-    /// signal on changed property of this object before document scoped signalChangedObject
-    boost::signals2::signal<void (const App::DocumentObject&, const App::Property&)> signalEarlyChanged;
+    struct Public;
+    Public* signals;  // pImpl

```

Comment:
I don't really think this is `pImpl`, though it is a nice pattern to reduce compile time.

How about:
```suggestion
    struct Singals;
    Signals* signals; // signals are indirecty accessed to allow faster compile time when not used
```